### PR TITLE
[CoW Subsidy] - Highlight current subsidy relative to CoW balance and show correct discount on SwapMod

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Part of #2549 

- Adds a hook: `useCowBalanceAndSubsidy` which returns cow balance (mocked) and the relevant subsidy info 
- Hook used to show correct fee discount in swapmod and in the modal
- adds the highlighting effect in table

Screenshots
image

Video of what it adds
https://user-images.githubusercontent.com/21335563/158879357-18b19280-bb2a-40fb-858b-dfb57dedb745.mov

### CODE REVIEW

File: src/custom/components/CowBalance/index.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface CowBalanceProps {
   account?: string
   balance?: CurrencyAmount<Currency>
   title?: string
+  subsidy: { tier: number; discount: number }
```
Comment (anxolin) [inline src/custom/components/CowBalance/index.tsx:18]: If you give a name to this type, you then can make it more readable below and not use `Pick<CowBalanceProps, 'subsidy'>`

**Review by anxolin — APPROVED** [review]

**Review by elena-zh — COMMENTED** [review]
Looks nice for a connected user (besides recently reported UI issues #2560 and #2561).
However, the level is not correct to a not connected user (seems to be that it should be 0). 
But it might be fixed in the next PRs...

**Comment by W3stside** [comment]
> Looks nice for a connected user (besides recently reported UI issues #2560 and #2561). However, the level is not correct to a not connected user (seems to be that it should be 0). But it might be fixed in the next PRs...

true! right now the balance is mocked :)

File: src/custom/components/CowSubsidyModal/utils.test.ts [hunk 2]
```diff
@@ -2,37 +2,45 @@
...
-  ['0', 0],
-  ['100', 10],
-  ['1000', 15],
-  ['10000', 25],
-  ['100000', 35],
+const BALANCES_IN_TIER = [
+  { balance: '0', tier: 0 },
+  { balance: '100000000000000000', tier: 0 }, // 0.1
+  { balance: '99990000000000000000', tier: 0 }, // 99.99
+  { balance: '100000000000000000000', tier: 0 }, // 100
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/utils.test.ts:12]: 100 should be tier1 already i believe

File: src/custom/components/CowSubsidyModal/utils.test.ts [hunk 3]
```diff
@@ -2,37 +2,45 @@
...
-      it(`USER BALANCE: [${MOCK_BALANCES[i]}] === TIER ${i}: [${
-        i ? '>' : ''
-      }${threshold}] => ${discount}% DISCOUNT`, () => {
-        const BALANCE_BN = new BigNumber(MOCK_BALANCES[i])
-        expect(getDiscountFromBalance(BALANCE_BN).discount).toEqual(discount)
+    BALANCES_IN_TIER.forEach(({ balance, tier }) => {
+      it(`USER BALANCE: [${formatUnits(balance, 18)}] equals TIER ${tier}`, () => {
+        const BALANCE_BN = new BigNumber(balance)
+
+        expect(getDiscountFromBalance(BALANCE_BN).tier).toEqual(tier)
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/utils.test.ts:35]: this is a breeze to review and maintain now, thx 🙏🙏🙏

File: src/custom/components/CowSubsidyModal/utils.ts [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
-    if (balanceAtomsBn.lt(new BigNumber(threshold))) break
+  // Here we use a sliced verison of our data without index 0 (0 amt tier)
+  // because loop-wise a balance less than or equal to 0 and 100 (indices 0 and 1, respectively) are the same
+  for (const [threshold, thresholdDiscount] of COW_SUBSIDY_DATA.slice(1)) {
+    // Increase our tier number only if we're not at the end of our list
+    if (tier < LAST_INDEX) {
+      // Is balance less than or equal to threshold?
+      // break. it's our tier.
+      const thresholdBn = new BigNumber(threshold)
+      if (balanceAtomsBn.lte(thresholdBn)) break
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/utils.ts:18]: `lt` since we credit always the next tier if the user is in the threshold.

Comment (anxolin) [inline src/custom/components/CowSubsidyModal/utils.ts:18]: if you use a `return` here your function is simpler. You can omit the `if (tier < LAST_INDEX)` and just return the last tier out of the loop

File: src/custom/components/CowSubsidyModal/utils.ts [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
+export function getDiscountFromBalance(balanceAtomsBn: BigNumber) {
   let discount = firstDiscount
   let tier = 0
-  for (const [threshold, thresholdDiscount] of COW_SUBSIDY_DATA) {
-    // is balance < current threshold?
-    // e.g 2443 < TIER 3: >1000? FALSE
-    if (balanceAtomsBn.lt(new BigNumber(threshold))) break
+  // Here we use a sliced verison of our data without index 0 (0 amt tier)
+  // because loop-wise a balance less than or equal to 0 and 100 (indices 0 and 1, respectively) are the same
+  for (const [threshold, thresholdDiscount] of COW_SUBSIDY_DATA.slice(1)) {
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/utils.ts:12]: should `COW_SUBSIDY_DATA.slice(1)` be already a constant? (so u don't need to reconstruct the array each time u ask for the tier. It doesn't change

**Review by anxolin — COMMENTED** [review]
Great changes

File: src/custom/components/CowSubsidyModal/utils.ts (hunk 5 above)
Comment (W3stside) [reply src/custom/components/CowSubsidyModal/utils.ts]: done

File: src/custom/components/CowSubsidyModal/utils.ts (hunk 4 above)
Comment (W3stside) [reply src/custom/components/CowSubsidyModal/utils.ts]: cool, done

Comment (W3stside) [reply src/custom/components/CowSubsidyModal/utils.ts]: yep, done

**Review by elena-zh — COMMENTED** [review]
Then approved!

**Pull request merged**